libheif/text.cc

```diff
@@ -0,0 +1,38 @@
+/*
+ * HEIF codec.
+ * Copyright (c) 2023 Dirk Farin <dirk.farin@gmail.com>
+ * Copyright (c) 2025 Brad Hards <bradh@frogmouth.net>
+ *
+ * This file is part of libheif.
+ *
+ * libheif is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version.
+ *
+ * libheif is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with libheif.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <string>
+#include "text.h"
+#include "error.h"
+#include <iostream>
+
+
+Error TextItem::encode(std::vector<uint8_t>& result) const
```

Comment:
You can pass back the result as a `Result<vector<uint8_t>>`.

Reply:
Again based on the `RegionItem` implementation - https://github.com/strukturag/libheif/blob/c6b98e5d9c2008cde501dfb0184ce89b46d985c2/libheif/region.cc#L118-L172

Reply:
I've switched the implementation to use this.